components/options_resolver.rst

```diff
@@ -367,6 +367,43 @@ to add additional allowed types without erasing the ones already set.
     the values to be given as an array mapping option names to allowed types:
     ``$resolver->setAllowedTypes(array('port' => array('null', 'int')));``
 
+You can specify an array of a specific type as an allowed option. The expected type is
+validated in the same way as before (``is_<type>()`` or an ``instanceof`` check).
+Only for an array, this is done recursively. If you expect an option to be an array of
```

Comment:
@ogizanagi Well the existing documentation does mention the validation is done using the `is_<type>` functions, but we could leave it out here as an _"implied implementation detail"_ or something.